lib/dal/src/socket/input.rs

```diff
@@ -37,10 +38,14 @@ pub enum InputSocketError {
     EdgeWeight(#[from] EdgeWeightError),
     #[error("func error: {0}")]
     Func(#[from] FuncError),
+    #[error("helper error: {0}")]
+    Helper(#[from] HelperError),
     #[error("layer db error: {0}")]
     LayerDb(#[from] LayerDbError),
     #[error("node weight error: {0}")]
     NodeWeight(#[from] NodeWeightError),
+    #[error("schema variant error: {0}")]
+    SchemaVariant(#[from] Box<SchemaVariantError>),
```

Comment:
I see the `Box` usages in the PR. I like trying to stick to one pattern (my preference is to use neither `Box` nor `String`, but to create smaller error types), but is there a chance of infinite recursion with error types even with using `Box`?

Reply:
String is not a good pattern, Box has no risk of infinite recursion. It will hold the depth just like String. We should not be using String as it removes context for no added safety.

This is a very common pattern for the rust community

Reply:
Thanks for the explanation!